assembl/static2/js/app/pages/voteSessionAdmin.jsx

```diff
@@ -93,35 +140,89 @@ class VoteSessionAdmin extends React.Component<void, VoteSessionAdminProps, void
         });
     }
 
-    if (modulesHaveChanged) {
-      const voteSession = voteSessionPage.toJS();
-      const vModules = voteModules.toJS();
-      const items = [
-        {
-          ...vModules[0],
-          voteSessionId: voteSession.id
-        }
-      ];
-      const mutationsPromises = getMutationsPromises({
-        items: items,
-        variablesCreator: createVariablesForTokenVoteSpecificationMutation,
-        deleteVariablesCreator: createVariablesForDeleteTokenVoteSpecificationMutation,
-        createMutation: createTokenVoteSpecification,
-        updateMutation: updateTokenVoteSpecification,
-        deleteMutation: deleteTokenVoteSpecification,
-        lang: i18n.locale
-      });
-
-      runSerial(mutationsPromises).then(() => {
-        refetchVoteSession();
-        displayAlert('success', I18n.t('administration.voteSessionSuccess'));
-      });
+    if (voteSessionPage.get('id')) {
+      if (tokenModulesHaveChanged) {
+        const tokenModules = voteModules.filter(m => m.get('type') === 'tokens');
+        const items = [];
+        tokenModules.forEach((t) => {
+          items.push({ ...t.toJS(), voteSessionId: voteSessionPage.get('id') });
+        });
+        const mutationsPromises = getMutationsPromises({
+          items: items,
+          variablesCreator: createVariablesForTokenVoteSpecificationMutation,
+          deleteVariablesCreator: createVariablesForDeleteMutation,
+          createMutation: createTokenVoteSpecification,
+          updateMutation: updateTokenVoteSpecification,
+          deleteMutation: deleteVoteSpecification,
+          lang: i18n.locale
+        });
+
+        runSerial(mutationsPromises).then(() => {
+          refetchVoteSession();
```

Comment:
Is this really necessary to refetch here? All the mutations should return the correct infos.